interpreter/cling/lib/Interpreter/Value.cpp

```diff
@@ -255,9 +264,43 @@ namespace cling {
     const clang::ASTContext& ctx = getASTContext();
     unsigned payloadSize = ctx.getTypeSizeInChars(getType()).getQuantity();
     m_Storage.m_Ptr = AllocatedValue::CreatePayload(payloadSize, dtorFunc,
-                                                    GetNumberOfElements());
+                                                GetNumberOfElements(getType()));
+  }
+
+  void Value::AssertTypeMismatch(const char* Type) const {
+    assert(isBuiltinType() && "Must be a builtin!");
+    const clang::BuiltinType *BT = getType()->castAs<clang::BuiltinType>();
+    clang::PrintingPolicy Policy = getASTContext().getPrintingPolicy();
+    (void) BT; (void) Policy;
```

Comment:
```suggestion
```
Given that the intent to use this/hide this behind `#ifndef NDEBUG` then those statement are counter-productive.  When removing the compiler would complain (about unused variables) if the function is use in the wrong context.